compiler/rustc_const_eval/messages.ftl

```diff
@@ -282,7 +281,7 @@ const_eval_pointer_out_of_bounds =
         *[many] bytes
     } starting at offset {$ptr_offset} is out-of-bounds
 const_eval_pointer_use_after_free =
-    pointer to {$allocation} was dereferenced after this allocation got freed
```

Comment:
Half-way through I realized that this says "dereferenced" even when it is just doing an in-bounds pointer arithmetic check. It was too late to easily split that fix into a separate commit -- sorry.